hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java

```diff
@@ -697,32 +698,36 @@ public static boolean scmInit(OzoneConfiguration conf,
         if (clusterId != null && !clusterId.isEmpty()) {
           scmStorageConfig.setClusterId(clusterId);
         }
-        scmStorageConfig.initialize();
-        LOG.info("SCM initialization succeeded. Current cluster id for sd={}"
-            + ";cid={};layoutVersion={}", scmStorageConfig.getStorageDir(),
-            scmStorageConfig.getClusterID(),
-            scmStorageConfig.getLayoutVersion());
-        //TODO: this check need not be done if ratis is not used.
-        // Initialize the raft group Id withe cluster Id.
+        // TODO: set the scm node info in the version file during upgrade
+        //  from non-HA SCM to SCM HA set up.
         if (SCMHAUtils.isSCMHAEnabled(conf)) {
           RaftServer server = SCMRatisServerImpl
               .newRaftServer(clusterId, scmStorageConfig.getScmId(), haDetails,
                   conf).build();
+          // ensure the ratis group exists
           server.start();
           server.close();
+          scmStorageConfig
```

Comment:
Just a question can we also set scmNodeInfo in non-HA also?
So that all fresh installs of non-HA also will have this info.

Reply:
Upgrade scenarios will need to be handled in case we want to add it for non-HA clsuter.

Reply:
Fine with me, If we want to revisit this later.

Reply:
Handling upgrades will be an issue as usual. Will revisit this later.